app/Models/Category.php

```diff
@@ -0,0 +1,22 @@
+<?php
+
+namespace App\Models;
+
+use Illuminate\Database\Eloquent\Factories\HasFactory;
+use Illuminate\Database\Eloquent\Model;
+
+class Category extends Model
+{
+    use HasFactory;
+
+    protected $fillable = [
+        'title',
+        'name',
+        'created_at',
+        'updated_at',
+    ];
+
+    public function posts() {
+        return $this->belongsToMany(Post::class,'category_posts','post_id','category_id');
+    }
```

Comment:
_If your pivot table name is **category_post** you can shorten it like the following sentence_

`return $this->belongsToMany(Post::class);`